### ISSUE TITLE
Someway to Sync a folder from remote?

### ISSUE THREAD

**Issue by jorgecuesta** [theophilusx/ssh2-sftp-client#171]
I'm looking for a way to keep a folder from remote in sync with destination. Thanks!

**Comment by theophilusx** [comment]
While ssh2-sftp-client does provide much of the functionality you would need to implement some sort of synchronization between local and remote folders, this is not a primary goal of the module, which is to provide the basic functionality to transfer data over the SFTP protocol. 

I have used this module to implement a file synchronization process. However, how this is done depends a lot on your individual requirements. For example, how closely they need to be synchronised i.e. every hour, every day, every week etc. The type of data also needs to be considered. For example, in the simplest case, you could use stat() calls to identify new files and files which have changed based on names, size, modified times etc. In other cases, something more sophisticated might be required, such as checksums. You may also need to build in some protection against race conditions - especially if the files are big. For example, if your checking for new files every 5 minutes, but your files are large enough or your network is slow enough that it might take more than 5 minutes for new files to be written to the source, how do you detect a file which has been completely transferred from one which is still being written to the remote source? 

IN summary, this module would certainly assist in writing something to implement the synchronization.   DEpending on your requirements, it could be quite trivial (see example folder in repo to see how easy it would be to transfer a directory of files). In other cases, your requirements might require significant additional functionality that you would either need to implement or find provided in a higher level module which possibly uses this module as one of it's dependencies.

**Comment by jorgecuesta** [comment]
Thanks @theophilusx for your answer, I finally go in that way doing me own logic on it. Regards!
